### PR TITLE
K8S-2511: ensure backward compatibility for mixed versions

### PR DESCRIPTION
**Pull request by tahmmee**
Applies defaults used in 2.3 to ensure validation passes when using upstream CRD's
on lower releases.

### CODE REVIEW

File: charts/couchbase-operator/values-all.yaml [hunk 1]
```diff
@@ -268,7 +269,7 @@ syncGateway:
   # -- image of the sync gateway container
   image:
     repository: couchbase/sync-gateway
-    tag: 2.8.0-enterprise
+    tag: 2.8.2-enterprise
```
Comment (patrick-stephens) [inline charts/couchbase-operator/values-all.yaml:272]: Should this be 2.8.3 now?

File: charts/couchbase-operator/values-all.yaml [hunk 2]
```diff
@@ -435,10 +436,12 @@ cluster:
...
   autoscaleStabilizationPeriod:
   # -- Backup defines whether the Operator should manage automated backups, and
-  # how to lookup backup resources.
+  # how to lookup backup resources.  Refer to the documentation for supported
+  # values https://docs.couchbase.com/operator/current/howto-backup.html#enable-
+  # automated-backup
   backup:
     # -- The Backup Image to run on backup pods
-    image: couchbase/operator-backup:1.0.0
+    image: couchbase/operator-backup:1.1.0
```
Comment (patrick-stephens) [inline charts/couchbase-operator/values-all.yaml:444]: Do we want to do any processing on these based on server version?

File: tools/value-generation/gen.py [hunk 3]
```diff
@@ -23,6 +23,7 @@ def format_properties(properties, values, comments, sub_keys, depth):
 
     # check for sub properties
     if 'properties' in value:
+
```
Comment (patrick-stephens) [inline tools/value-generation/gen.py:26]: Why the empty line?

File: tools/value-generation/gen.py [hunk 4]
```diff
@@ -102,11 +139,12 @@ def processCluster(crd_value, value_map, comment_map) :
     if expectedKey not in value_map[crd_value]:
       value_map[crd_value][expectedKey] = {}
 
-  value_map[crd_value]['backup']['image'] = 'couchbase/operator-backup:1.0.0'
+  value_map[crd_value]['backup']['image'] = 'couchbase/operator-backup:1.1.0'
   value_map[crd_value]['backup']['managed'] = True
 
   value_map[crd_value]['buckets']['managed'] = True
   value_map[crd_value]['image'] = 'couchbase/server:6.6.2'
```
Comment (patrick-stephens) [inline tools/value-generation/gen.py:146]: Should we be setting to 7.0.2 with the scope processing as well? Probably not to align with 2.2.1.

Might want to consider 6.6.3 though?

File: tools/value-generation/gen.py [hunk 5]
```diff
@@ -134,6 +172,9 @@ def processCluster(crd_value, value_map, comment_map) :
   # Set this empty to ensure we auto-generate it by default
   value_map[crd_value]['security']['adminSecret'] = ''
 
+  # Set monitoring to emtpy since it requires image which is not provided by default
```
Comment (patrick-stephens) [inline tools/value-generation/gen.py:175]: And for 7.0.2 we can use the server image to provide it.

**Review by patrick-stephens — APPROVED** [review]

File: tools/value-generation/gen.py (hunk 4 above)
Comment (tahmmee) [reply tools/value-generation/gen.py:146]: 6.6.3 sounds legit

File: tools/value-generation/gen.py (hunk 5 above)
Comment (tahmmee) [reply tools/value-generation/gen.py:175]: Indeed. for 2.2.3 this will be the move

File: charts/couchbase-operator/values-all.yaml (hunk 1 above)
Comment (tahmmee) [reply charts/couchbase-operator/values-all.yaml:272]: yes

File: charts/couchbase-operator/values-all.yaml (hunk 2 above)
Comment (tahmmee) [reply charts/couchbase-operator/values-all.yaml:444]: Not sure, what's the suggestion?)

Comment (patrick-stephens) [reply charts/couchbase-operator/values-all.yaml:444]: I think we are ok at the moment but there was a point where you had to use backup image X with server version Y I thought. We probably just need to make sure the defaults work though.

**Pull request merged**